### PR TITLE
Support numeric public key ids for actions

### LINKED ISSUE #1835: GitHub Action public key key ids are returned as numeric in some versions of GitHub Enterprise
Tested using GitHub Enterprise  Server 2.22.6, I'm having a problem creating secrets with the terraform provider (https://github.com/integrations/terraform-provider-github/blob/master/github/data_source_github_actions_public_key.go#L39) because go is unable to unmarshal a numeric value into a string.

I'll be pushing a pull request shortly.

### PR DESCRIPTION
**Pull request by Azuka**
closes #1835

### CODE REVIEW

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:38]: Why not just get rid of line 37?

```suggestion
		p.KeyID = String(strconv.FormatFloat(v, 'f', -1, 64))
```

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:21]: This method needs a godoc comment describing its use.

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:28]: This line should be covered in a unit test that provides it with garbage data.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @Azuka .
Please address the comments below.

File: github/actions_secrets.go
Comment (Azuka) [reply github/actions_secrets.go]: That was pretty obvious. Thanks!

**Comment by Azuka** [comment]
@gmlewis thanks for reviewing. I've made the suggested changes

File: github/actions_secrets.go
Comment (gmlewis) [inline github/actions_secrets.go:23]:
```suggestion
// do not error out when unmarshaling.
```

File: github/actions_secrets_test.go
Comment (gmlewis) [inline github/actions_secrets_test.go:24]: Can you please add four more test cases?

```go
"Nil": {
  data: nil,
  ...
},
"Empty String": {
  data: []byte(""),
  ...
},
"Missing Key": {
  data: []byte(`{"key_id":"1234"}`),
  ...
},
"Missing KeyID": {
  data: []byte(`{"key":"2Sg8iYjAxxmI2LvUXpJjkYrMxURPc8r+dB7TJyvv1234"}`),
  ...
},
```

File: github/actions_secrets_test.go
Comment (gmlewis) [inline github/actions_secrets_test.go:40]: This test case says "Numeric KeyID" but the "key_id" field is not numeric.  Could you please fix this?

File: github/actions_secrets_test.go
Comment (gmlewis) [inline github/actions_secrets_test.go:35]: This test case says "String KeyID" but the "key_id" field is not a string.  Could you please fix this?

**Review by gmlewis — CHANGES_REQUESTED** [review]

**Comment by gmlewis** [comment]
> I'm really sorry, I missed that.

No problem.  Next time.  :joy:

**Review by gmlewis — APPROVED** [review]
Thank you, @Azuka !
LGTM.

Awaiting second LGTM before merging.

**Review by wesleimp — APPROVED** [review]
💙

**Comment by gmlewis** [comment]
Thank you, @wesleimp .
Merging.

**Pull request merged**
